### ISSUE TITLE
Run PlannedReparentShard if primary vttablet is shutting down

### ISSUE THREAD

**Issue by L3o-pold** [planetscale/vitess-operator#276]
If a primary vttablet is shutting down, it could be nice to run a `PlannedReparentShard` command.

It could occurs in multiple scenario where the user may or may not be able to run it by himself. It could also avoid a mistake from the user where he forgot to run it before shutting down the vttablet.

Those scenario could be:
- Node drain
- Pod eviction
- other?

**Comment by frouioui** [comment]
Hi @L3o-pold, in which situation would you shutdown a vttablet? In case of an error or in case you're scaling down your replica set?

**Comment by L3o-pold** [comment]
Hi @frouioui, node drain is the most common operation for us. And the team in charge of Kubernetes cluster is not the same as the data team so it requires more procedure than necessary IMHO

**Comment by frouioui** [comment]
You have some vttablets (including maybe a primary) in a node and you may drain that node, in which case you would want to automatically do one or more PRS before draining the node if there are some primary tablets, am I getting this right?

I need to try this out to understand how vtop behaves when the user drains a node, but it sounds like something we should do eventually.

**Comment by L3o-pold** [comment]
You are correct. Today, our platform / Kubernetes teams needs to be aware that the Kubernetes cluster is hosting a Vitess cluster, and needs to run a custom script to drain node to do this operation (PRS).

Draining a node is common for maintenance operation (upgrade, debug, maintenance).

You can try it out by running `kubectl drain ${name_of_your_node}` :)

**Comment by bluecrabs007** [comment]
We worked around this exact issue running a manual [drain](https://github.com/planetscale/vitess-operator/blob/main/pkg/controller/vitessshardreplication/reconcile_drain.go) on the pod before we drained the node.
By applying the following annotation on the vttablet pod, the pod is PRS'ed and we then proceed with `kubectl drain`
```
kubectl patch pod $pod --type json  --patch='[ {"op": "add", "path": "/metadata/annotations/drain.planetscale.com~1started", "value": "foo bar"} ]'
```

If the pod has to PRS itself before it gets evicted by the `kubectl drain`, it will have to block all containers from exiting, using a preStop hook or something like that and signal the vtop for a PRS.
